src/uwtools/api/config.py

```diff
@@ -28,24 +33,50 @@ def compare(
 
 
 def realize(
-    input_file: OptionalPath,
-    input_format: str,
-    output_file: OptionalPath,
-    output_format: str,
-    values_file: OptionalPath,
-    values_format: Optional[str],
+    input_config: Union[dict, _Config, OptionalPath] = None,
+    input_format: Optional[str] = None,
+    output_file: OptionalPath = None,
+    output_format: Optional[str] = None,
+    values: Union[dict, _Config, OptionalPath] = None,
+    values_format: Optional[str] = None,
     values_needed: bool = False,
     dry_run: bool = False,
 ) -> bool:
+    """
+    ???
+    """
+    realize_to_str(
+        input_config=input_config,
+        input_format=input_format,
+        output_file=output_file,
+        output_format=output_format,
+        values=values,
+        values_format=values_format,
+        values_needed=values_needed,
+        dry_run=dry_run,
+    )
+    return True
```

Comment:
The `realize()` API function calls `realize_to_str()` and just discards the returned string value. It needs to return a `bool` to interface correctly with the CLI. If errors occur in the library, past the `realize()` call, they should raise exceptions rather than signaling errors via return values.

Sorry about all the boilerplate: Much of what's happening in the API code is just piping calls from the CLI or users to the `uwtools` innards. We'll add value later to these blocks by writing good docstrings that will be turned into HTML API documentation for users.